common.mk

```diff
@@ -9,7 +9,7 @@ SHELL=/bin/bash
 lookup_scala_srcs = $(shell find -L $(1)/ -iname "*.scala" 2> /dev/null)
 
 PACKAGES=rocket-chip testchipip boom hwacha sifive-blocks
-SCALA_SOURCES=$(foreach pkg,$(PACKAGES),$(call lookup_scala_srcs,$(base_dir)/generators/$(pkg)/src/main/scala)) $(call lookup_scala_srcs,$(base_dir)/src/main/scala)
+SCALA_SOURCES=$(foreach pkg,$(PACKAGES),$(call lookup_scala_srcs,$(base_dir)/generators/$(pkg)/src/main/scala)) $(call lookup_scala_srcs,$(base_dir)/example/src/main/scala) $(call lookup_scala_srcs,$(base_dir)/utilities/src/main/scala)
```

Comment:
Maybe we should move them to `generators` so that you can just put `example` and `utilities` in the `PACKAGES` list?

Reply:
I think we want to have a distinction between what are the base projects and what is stitching them together. I personally, this is fine despite it being a bit long.

Reply:
Why do we want this distinction? Isn't it evident from the git submodules? Would you be ok with `PACKAGES=$(BASE_PACKAGES) $(SUBMODULE_PACKAGES)`?

Reply:
The `generators` directory is really a "ip" directory, so it feels wrong to move `example` and `utilities` there.

Reply:
I think co-locating these two in some other sub-directory fits with the other organization decisions.

Reply:
You could have moved to some other directory but I personally like this :)